packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts

```diff
@@ -448,9 +448,11 @@ export async function askAdditionalAuthQuestions(context, authConfig, defaultAut
       authConfig.additionalAuthenticationProviders.push(config);
     }
   } else {
-    authConfig.additionalAuthenticationProviders = (currentAuthConfig.additionalAuthenticationProviders || []).filter(
-      p => p.authenticationType !== defaultAuthType,
-    );
+    if (currentAuthConfig) {
+      authConfig.additionalAuthenticationProviders = (currentAuthConfig.additionalAuthenticationProviders || []).filter(
```

Comment:
could use `currentAuthConfig?.additionalAuthenticationProviders` instead of the if statement